### PR TITLE
[Term Entry] Kotlin Datetime .asTimeZone() 

### LINKED ISSUE #3805: [Term Entry] Kotlin Datetime .asTimeZone()
### Reason/inspiration (optional)

We would like a new term entry on the `.asTimeZone()` method for Kotlin: [Datetime](https://www.codecademy.com/resources/docs/kotlin/datetime). The entry should go in a new file under `docs/content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```kotlin `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Kadif-Ravik**
### Description

Term entry for Kotlin datetime, .asTimeZone()

Closes 3805

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+fun UtcOffset.asTimeZone(): FixedOffsetTimeZone
+
+```
+
+`TimeZone` and `FixedOffsetTimeZone` data  can be converted between `Instant` and `LocalDate`.
+
+## Example
+
+In the following example a`TimeZone` object, `timeZone` with the ID `America/ Los_Angeles` is created. The `.asTimeZone()` function converts the `TimeZone` object to a string  representation of the time zone and then printed to the console.
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md:31]:
```suggestion
In the following example a `TimeZone` object, `timeZone` with the ID `America/ Los_Angeles` is created. The `.asTimeZone()` function converts the `TimeZone` object to a string representation of the time zone.
```

File: content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+print(asString)
+
+```
+
+The output will be in the format "UTC offset +/- HH:MM.
+
+```shell
+
+-08:00
+
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md:46]:
```suggestion
-08:00
```
Add the proper and accurate output.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Kadif-Ravik Here are a few changes from my side, I might suggest some more changes in the future.

**Comment by Kadif-Ravik** [comment]
@avdhoottt I'm having trouble with the main function.

File: content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---
+Title: '.asTimeZone'
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md:2]:
```suggestion
Title: '.asTimeZone()'
```

File: content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+```kotlin
+import kotlinx.datetime*
+import kotlinx.serializer*
+
+fun main() {
+  val timeZone = TimeZone.getTimeZone("America/Los_Angeles")
+  val asString = timeZone.asTimeZone()
+  print(asString)
+}
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md:37]:
```suggestion
import kotlinx.datetime.*

fun main() {
  // Create a UTCOffset object with an offset of +5 hours, 30 minutes
  val utcOffset = UtcOffset(5, 30)
  val fixedOffsetTimeZone: FixedOffsetTimeZone = utcOffset.asTimeZone()
  println("$fixedOffsetTimeZone") 
}
```
I've provided a code in Kotlin, please have a look at it. I think the `.asTimeZone()` method only works on UTC offset, so I've created one and called the method on it.

File: content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+The output will be
+
+```shell
+-08:00
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md:43]:
```suggestion
+05:30
```
Updated the output as per the code.

File: content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+```pseudo
+fun UtcOffset.asTimeZone(): FixedOffsetTimeZone
+```
+
+`TimeZone` and `FixedOffsetTimeZone` data can be converted between `Instant` and `LocalDate`.
+
+## Example
+
+In the following example a `TimeZone` object, `timeZone` with the ID `America/ Los_Angeles` is created. The `.asTimeZone()` function converts the `TimeZone` object to a string representation of the time zone.
```
Comment (avdhoottt) [inline content/kotlin/concepts/datetime/terms/asTimeZone/asTimeZone.md:27]: Please review the code and change the description accordingly.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
@Kadif-Ravik I have looked into the code and have updated the code as per my knowledge. Please have a look at it and make the suggested changes.

**Comment by avdhoottt** [comment]
@Kadif-Ravik Are you still working on this PR? Please make the suggested changes.

**Comment by ishg-153** [comment]
@avdhoottt You should consider closing this PR and opening up the corresponding issue so that someone else can work on it.

**Comment by ishg-153** [comment]
Closing this PR and opening up its corresponding issue.

@Kadif-Ravik @avdhoottt

**Pull request closed without merging**